route_following_plugin/include/route_following_plugin.h

```diff
@@ -121,20 +103,31 @@ namespace route_following_plugin
          */
         bool identifyLaneChange(lanelet::routing::LaneletRelations relations, int target_id);
 
+        void set_maneuver_start_dist(cav_msgs::Maneuver& maneuver, double start_dist);
+
         /**
          * \brief Service callback for arbitrator maneuver planning
          * \param req Plan maneuver request
          * \param resp Plan maneuver response with a list of maneuver plan
          * \return If service call successed
          */
         bool plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp);
+
+        double get_final_speed(cav_msgs::Maneuver maneuver);
```

Comment:
Let's also make sure to doc-comment these new members before we close this out.

Reply:
These functions were made public so they can be individually unit tested

Reply:
I'd recommend using the friend test pattern like this: https://stackoverflow.com/questions/2396370/how-to-make-google-test-classes-friends-with-my-classes That should let the unit tests access the private/protected members of the classes without making them public in a broader sense. Either that or it might indicate that these functions might be worth factoring out into a separate class, from the perspective of the responsibilities of the route_following plugin I'm not sure they make sense to be part of the external API.